src/pages/ProductDetail/ProductDetail.js

```diff
@@ -1,5 +1,578 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Link, useNavigate, useParams } from 'react-router-dom';
+import ImgArea from './ImgArea/ImgArea';
+import './ProductDetail.scss';
 
 export default function ProductDetail() {
-  return <div>ProductDetail</div>;
+  // 상품 전체 데이터
+  const [productData, setProductData] = useState({});
+  //수량갯수
+  const [quantityNum, setQuantityNum] = useState(1);
+  const [postSize, setPostSize] = useState();
+  const [checkSize, setCheckSize] = useState('사이즈 선택');
+
+  //
+  const params = useParams();
+  const productId = params.productId;
+
+  const navigate = useNavigate();
+
+  // 서버연결용
+  // useEffect(() => {
+  //   fetch(`http://10.58.52.205:3000/product-detail/${productId}`, {
+  //     method: 'GET',
+  //     headers: { 'Content-Type': 'application/json;charset=utf-8' },
+  //   })
+  //     .then(response => response.json())
+  //     .then(result => {
+  //       setProductData(result.data[0]);
+  //     });
+  // }, []);
+
+  // 로컬용
+  useEffect(() => {
+    fetch('/data/productDetailMockData.json')
+      .then(response => response.json())
+      .then(result => {
+        setProductData(result.data[0]);
+      });
+  }, []);
+
+  //수량차감함수
+  function quantityMinus() {
+    let num = quantityNum;
+    if (num === 1) {
+      setQuantityNum(1);
+    } else {
+      setQuantityNum(num - 1);
+    }
```

Comment:
재할당되는 변수가 아니라면 let을 사용하실 필요 없습니다!
또한, 따로 변수로 선언할 필요 없이 state 자체 혹은 함수형 업데이트로 작성할 수 있겠네요!
```suggestion
    if (num === 1) return;
    
    setQuantityNum(quantityNum - 1);
```
```suggestion
    if (num === 1) return;
    
    setQuantityNum(prev => prev - 1);
```

Reply:
setState의 인자로 사용된 함수는 이전 state값을 전달받으며 그 값을 이용한 함수들은 큐에 저장되어 순서대로 실행...
적용완료!